File: Python/beginner/1534.py
# https://judge.beecrowd.com/pt/problems/view/1534
# Matriz 123

# Loop to read n until EOF
while True:
    try:
        n = int(input())
    except EOFError:
        break
    except ValueError:
        continue

    # Loop to print the matrix
    for i in range(n):
        for j in range(n):
            if i == n - 1 - j: # Needs to be evaluated first for find correct positions when an element attens condition to be on both diagonals
                print(2, end='')
            elif i == j: # Main Diagonal
                print(1, end='')
            else: # Other positions
                print(3, end='')
        print()
    
# By: Fernando Serra
# https://github.com/fernandosserra

File: Haskell/beginner/1534.hs
-- https://judge.beecrowd.com/pt/problems/view/1534
-- Matriz 123

-- Importing replicateM and printf
import Control.Monad (replicateM)
import Text.Printf (printf)

-- Function to calculate if the value are in the main diagonal, in the secondary diagonal or in the other positions
positionFinder :: Int -> Int -> Int -> Int
positionFinder r c n
    | r + c == n - 1 = 2  -- Secondary diagonal (needs to be evaluated first)
    | r == c         = 1  -- Main diagonal 
    | otherwise      = 3  -- Other positions

-- Main Function
main :: IO ()
main = do
    -- Loop to read multiple entries until EOF
    contents <- getContents
    mapM_ processLine $ lines contents
  where
    processLine line = do
        let n = read line :: Int
        -- Generating our Matrix line by line
        let matrix = [[positionFinder r c n | c <- [0..n-1]] | r <- [0..n-1]]
        -- Printing the matrix
        mapM_ (putStrLn . concatMap show) matrix

-- By: Fernando Serra
-- https://github.com/fernandosserra